### PR TITLE
DES-68 - Create JS object to be passed into bootstrap

### PR DESCRIPTION
**Pull request by GHaberis**
**Initialise templates with JS obj instead of having to query body elem for template info**
- Add javascript object to templates which is passed to new method GU.Bootstrap.init

**Take objects out of global namespace and add to GU namespace**
- eg. window.initTemplate is now GU.Bootstrap.init

**Update window.requestAnimationFrame shim with actual polyfill**

**Write unit tests for app.js**
- Make app.js testable (replace anonymous functions)

**Delete all old test fixtures (outdated) and replace with new fixtures**
Most of the file changes are simply deleting old static HTML fixtures and adding new ones, these won't need reviewing.

### CODE REVIEW

**Comment by GuardianAndroid** [comment]
SonarQube has found a total of 0 issues.


For a CSS diff, check the wraith gallery [here](http://iosuiauto.gutools.co.uk:8080/view/Mobile%20Templates/job/TemplatesPullRequest/175/CSS_Diff/).

**Comment by GuardianAndroid** [comment]
SonarQube has found a total of 0 issues.


For a CSS diff, check the wraith gallery [here](http://iosuiauto.gutools.co.uk:8080/view/Mobile%20Templates/job/TemplatesPullRequest/176/CSS_Diff/).

**Comment by GHaberis** [comment]
@guardian/dotcom-apps-collab Hi, would someone be willibg to review this pull request for the app templates? The main changes are listed in the initial comment I made. Thanks!

**Comment by GuardianAndroid** [comment]
SonarQube has found a total of 0 issues.


For a CSS diff, check the wraith gallery [here](http://iosuiauto.gutools.co.uk:8080/view/Mobile%20Templates/job/TemplatesPullRequest/177/CSS_Diff/).

File: ArticleTemplates/audioTemplate.html [hunk 1]
```diff
@@ -85,7 +88,24 @@ <h1 class="headline selectable card__title--audio"><span class="card__titletext"
     __COMMENTS_MODULE__
 
     <script type="text/javascript">
-        window.initTemplate();
+        GU.Bootstrap.init({
```
Comment (gtrufitt) [inline ArticleTemplates/audioTemplate.html]: Minor, but is there the concept of partials so you don't have to update this across all templates anytime you add or change anything?

File: ArticleTemplates/commentsTemplate.html [hunk 2]
```diff
@@ -47,7 +50,18 @@ <h1 class="comments__headline headline">__TITLE__</h1>
             </div>
         </section>
         <script type="text/javascript">
-            window.initTemplate({ skipStyle: true });
+            GU.Bootstrap.init({
+                sectionTone: '__SECTION_TONE__',
```
Comment (gtrufitt) [inline ArticleTemplates/commentsTemplate.html]: Although looks like they're slightly different in each template

File: ArticleTemplates/interactiveTemplate.html [hunk 3]
```diff
@@ -6,15 +6,32 @@
...
+    <script type="text/javascript">
+        window.GU = window.GU || {};
+    </script>
     <script type="text/javascript" src="__TEMPLATES_DIRECTORY__assets/build/shim.js"></script>
 </head>
 
-<body class="__FONT_SIZE__ __PLATFORM__ __IS_OFFLINE__" data-fullscreen-interactive data-content-type="article" data-ads-enabled="__ADS_ENABLED__" data-ads-enable-hiding="__ADS_ENABLE_HIDING__" data-ads-config="__ADS_CONFIG__" style="margin-top: __ACTIONBARHEIGHT__px;" data-font-size="__FONT_SIZE_INT__" data-page-id="__PAGE_ID__" data-template-directory="__TEMPLATES_DIRECTORY__" data-page-id="__PAGE_ID__" data-mpu-after-paragraphs="__MPU_AFTER_PARAGRAPHS__" data-use-ads-ready="__ADS_FAST_CALLBACK__">
+<body class="__FONT_SIZE__ __PLATFORM__ __IS_OFFLINE__" data-fullscreen-interactive data-content-type="interactive" data-ads-enabled="__ADS_ENABLED__" data-ads-enable-hiding="__ADS_ENABLE_HIDING__" data-ads-config="__ADS_CONFIG__" style="margin-top: __ACTIONBARHEIGHT__px;" data-font-size="__FONT_SIZE_INT__" data-page-id="__PAGE_ID__" data-template-directory="__TEMPLATES_DIRECTORY__" data-mpu-after-paragraphs="__MPU_AFTER_PARAGRAPHS__" data-use-ads-ready="__ADS_FAST_CALLBACK__">
```
Comment (gtrufitt) [inline ArticleTemplates/interactiveTemplate.html]: Assuming the ultimate goal is to remove all these data attributes?

**Comment by gtrufitt** [comment]
👍 Bar minor comments, looks good!

File: ArticleTemplates/interactiveTemplate.html (hunk 3 above)
Comment (GHaberis) [inline ArticleTemplates/interactiveTemplate.html]: Yep definitely! I'm leaving them in there for now as they're queried all over the place. I didn't want to bloat the scope of this ticket by searching and replacing these queries (I only did it in app.js), however I will raise another ticket to do this.

**Comment by GHaberis** [comment]
@gtrufitt thanks for looking over it!

**Pull request merged**